tokio/src/sync/rwlock.rs

```diff
@@ -158,6 +467,15 @@ unsafe impl<T> Send for RwLock<T> where T: Send {}
 unsafe impl<T> Sync for RwLock<T> where T: Send + Sync {}
 unsafe impl<'a, T> Sync for RwLockReadGuard<'a, T> where T: Send + Sync {}
 unsafe impl<'a, T> Sync for RwLockWriteGuard<'a, T> where T: Send + Sync {}
+// NB: These impls need to be explicit since we're storing a raw pointer.
+// Safety: Stores a raw pointer to `T`, so if `T` is `Sync`, so is the lock
+// guard over `T`.
+unsafe impl<'a, T> Send for MappedRwLockReadGuard<'a, T> where T: Sync {}
```

Comment:
Should these be implementing `Sync`, not `Send`? I don't know for sure, but it's suspicious given the code around it and the comment above it.

Reply:
Modified the comment. The impl is indeed supposed to be `Send` if we want the guard to be sent across `.await` for `Send` futures. See if it makes sense to you now.